### PR TITLE
Escaping glob patterns from context. Issue #132

### PR DESCRIPTION
**Pull request by NRosewood**
This PR contains a:

- [x] **bugfix**
- [ ] new **feature**
- [ ] **code refactor**
- [x] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case

If you do not escape glob patterns from context, then the plugin will not match files for linting.

I described the problem in detail in [Issue #132](https://github.com/webpack-contrib/eslint-webpack-plugin/issues/132)

### Additional Info
Also, this is not the only problem that square brackets are causing. At least that's what the tests say.
If, again, there are brackets in the name of one of the directories before the project, then most of the tests will fail.

### CODE REVIEW

**Comment by alexander-akait** [comment]
Make sense to look at stable solutions from `fast-glob` https://github.com/mrmlnc/fast-glob/blob/master/src/utils/path.ts

File: test/symbols.test.js [hunk 1]
```diff
@@ -10,8 +10,11 @@ describe('symbols', () => {
   it('should return error', (done) => {
     const compiler = pack(
       'symbols',
-      {},
-      { context: join(__dirname, 'fixtures/[symbols]') }
```
Comment (alexander-akait) [inline test/symbols.test.js:14]: Keep this test, just add new

Comment (NRosewood) [reply test/symbols.test.js:14]: This test is the point of the issue, I guess.
I think so because this one was added after #93 as check for problem

**Comment by NRosewood** [comment]
> Make sense to look at stable solutions from `fast-glob` https://github.com/mrmlnc/fast-glob/blob/master/src/utils/path.ts#L19

In the issue, I described the reason why escaping a double backslash will not work. Backslashes will be removed after [normalizePath](https://github.com/webpack-contrib/eslint-webpack-plugin/blob/master/src/utils.js#L57)

**Comment by alexander-akait** [comment]
It means we should do normalize after escaping

**Comment by NRosewood** [comment]
`С:\\path\\[glob-pattern]\\project`
after fast-glob escape will turn to 
`С:\\path\\[glob-pattern\\]\\project (already broken)`
and then after normalizePath:
`С:/path/[glob-pattern/]/project/`

**Comment by alexander-akait** [comment]
I see, windows characters are not allowed in glob, so we should convert them to glob style firstly

**Comment by NRosewood** [comment]
I do not see a way to do this.
The problem lies here: [utils.js#L57](https://github.com/webpack-contrib/eslint-webpack-plugin/blob/master/src/utils.js#L57). 
We resolving context and file paths. Which returns windows path style again.
i.e. it doesn't matter what we do with context. Even if we convert context before, it will be with windows backslashes again and glob escape will not work

**Comment by alexander-akait** [comment]
Just prevent it

**Comment by NRosewood** [comment]
How to prevent make path methods use backslashes? In docs for node path i didn't found any solution

**Comment by alexander-akait** [comment]
You have regexps to fix it

**Comment by NRosewood** [comment]
Can you show me pseudo-code example? Where exactly do you think i should use regexp to fix issue

**Comment by alexander-akait** [comment]
Do it on `compiler.options.context`, so we will have valid `context`, when use `/` to concat with `glob` + we need escape special character, just avoid using `path`, also we need add tests with aboslute and relative paths to ensure we don't break them here

**Comment by NRosewood** [comment]
This idea is overcomplicated if I got you right
In this case we will need to write own path resolving function, which also may work incorrectly in some non-obvious cases

**Comment by NRosewood** [comment]
I dunno, maybe it would be better to slice context for `file` and `wanted/exclude` before put it in `isMatch`?
https://github.com/webpack-contrib/eslint-webpack-plugin/blob/master/src/index.js#L112

**Comment by alexander-akait** [comment]
We have this logic in `copy-webpack-plugin`, where you can use glob, you can look at code

**Pull request closed without merging**